app/models/image.rb

```diff
@@ -10,7 +10,7 @@ class Image < ApplicationRecord
   validates_lengths_from_database
   validates :username, :operatingsystem_id, :compute_resource_id, :architecture_id, :presence => true
   validates :name, :presence => true, :uniqueness => {:scope => [:compute_resource_id, :operatingsystem_id]}
-  validates :uuid, :presence => true, :uniqueness => {:scope => :compute_resource_id}
+  validates :uuid, :presence => true, :uniqueness => {:scope => :compute_resource_id}, :format => URI::regexp(%w(http https ftp))
```

Comment:
Note, that this differs per compute resource provider. This might be an actual uuid.
It's probably best to offload this to the compute resource, e.g. ask it what format it uses for uuids.
I'd check how we validate the uuid for hosts.

Reply:
Yeah this was my first thought too. We need to test this per CR basis. It's quite challenging patch actually, lot of work to do. Thanks for doing that! ;-)

Reply:
Created new Redmine issue for this as https://projects.theforeman.org/issues/25081 , as the issue was originally related to Hammer, but it seems to be there in Foreman too. :)